collisions_functions.py

```diff
@@ -0,0 +1,197 @@
+"""This module defines atom class and functions used in an elastic
+collision physics simulation.
+"""
+
+import math
+import random
+import numpy
+from random import randrange, uniform
+
+
+# Parameters
+red = (255,0,0)
+
+
+class Atom:
+    """ Defines an atom structure. """
+    def __init__(self, rad, color, max_vel, boundary):
+        self.ID = -1
+        self.rad = rad
+        self.mass = self.rad**2 * math.pi
+        self.color = color
+        self.vel = [uniform(-max_vel, max_vel), uniform(-max_vel, max_vel)] 
+        self.x = randrange(boundary[0] + self.rad + math.ceil(max_vel),
+                           boundary[1] - self.rad - math.ceil(max_vel))
+        self.y = randrange(boundary[2] + self.rad + math.ceil(max_vel),
+                           boundary[3] - self.rad - math.ceil(max_vel))
+
+
+def add_atom(atoms, sections, new_atom):
+    """Takes a list of atoms and checks if a new atom overlaps anything.
+    new_atom is appended to the atoms list if no overlaps.
+    """
+    for i in range(len(atoms)):
+        dist = ((new_atom.x-atoms[i].x)**2 + (new_atom.y-atoms[i].y)**2)**0.5
+        if dist <= new_atom.rad + atoms[i].rad:
+            return False
+
+    color_atoms(new_atom, red)
+    new_atom.ID = len(atoms) + 1
+    atoms.append(new_atom)
+
+    return True
+
+
+def color_atoms(atom, color):

```

Comment:
I notice that a lot of these functions take an `atom` as the first parameter.  I have a slight feeling that the idiomatic way to do this in python would be to make these _instance methods_ of the `Atom` class, which would look like

```
class Atom:
    def color(self, color):
        if self.x < 500:
            self.color = color
```

and then it could be called as `atom.color(color)`.  (It could technically still be called as `color(atom, color)` as well, but that would be an unusual way to invoke an instance method.)

Reply:
That makes a lot of sense, thanks!